components/script/dom/gpuqueue.rs

```diff
@@ -116,33 +116,45 @@ impl GPUQueueMethods for GPUQueue {
         data_offset: GPUSize64,
         size: Option<GPUSize64>,
     ) -> Fallible<()> {
-        let bytes = match data {
+        // Step 1
+        let sizeof_element: usize = match data {
+            BufferSource::ArrayBufferView(ref d) => d.get_array_type().byte_size().unwrap_or(1),
+            BufferSource::ArrayBuffer(_) => 1,
+        };
+        let data = match data {
             BufferSource::ArrayBufferView(d) => d.to_vec(),
             BufferSource::ArrayBuffer(d) => d.to_vec(),
         };
+        // Step 2
+        let data_size: usize = data.len() / sizeof_element;
+        assert_eq!(data.len() % sizeof_element, 0);
```

Comment:
If `byte_size` is wrong (this is maintained by us), this can be wrong too, so I would prefer to keep assert, although I will change it to debug assert.